tsfile/src/main/java/org/apache/iotdb/tsfile/fileSystem/IoTDBFile.java

```diff
@@ -0,0 +1,183 @@
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iotdb.tsfile.fileSystem;
+
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.*;
+import org.apache.iotdb.tsfile.common.conf.TSFileConfig;
+
+import java.io.File;
+import java.io.FileFilter;
+import java.io.IOException;
+import java.net.URI;
+import java.util.ArrayList;
+
+public class IoTDBFile extends File {
+
+  private Path hdfsPath;
+
+  public IoTDBFile(String pathname) {
+    super(pathname);
+    if (TSFileConfig.isHdfsStorage) {
+      hdfsPath = new Path(pathname);
+    }
+  }
+
+  public IoTDBFile(String parent, String child) {
+    super(parent, child);
+  }
+
+  public IoTDBFile(File parent, String child) {
+    super(parent, child);
+  }
+
+  public IoTDBFile(URI uri) {
+    super(uri);
+  }
+
+  @Override
+  public String getAbsolutePath() {
+    if (TSFileConfig.isHdfsStorage) {
+      return hdfsPath.toUri().toString();
+    }
+    return super.getAbsolutePath();
+  }
+
+  @Override
+  public long length() {
+    if (TSFileConfig.isHdfsStorage) {
+      try {
+        FileSystem fs = hdfsPath.getFileSystem(new Configuration());
+        return fs.getFileStatus(hdfsPath).getLen();
+      } catch (IOException e) {
+        e.printStackTrace();
+      }
+    }
+    return super.length();
+  }
+
+  @Override
+  public boolean exists() {
+    if (TSFileConfig.isHdfsStorage) {
+      try {
+        FileSystem fs = hdfsPath.getFileSystem(new Configuration());
+        return fs.exists(hdfsPath);
+      } catch (IOException e) {
+        e.printStackTrace();
+      }
+    }
+    return super.exists();
+  }
+
+  public IoTDBFile[] listFiles() {
+    if (TSFileConfig.isHdfsStorage) {
+      ArrayList<IoTDBFile> files = new ArrayList<>();
+      try {
+        FileSystem fs = hdfsPath.getFileSystem(new Configuration());
+        RemoteIterator<LocatedFileStatus> iterator = fs.listFiles(hdfsPath, true);
+        while (iterator.hasNext()) {
+          LocatedFileStatus fileStatus = iterator.next();
+          Path fullPath = fileStatus.getPath();
+          files.add(new IoTDBFile(fullPath.toUri()));
+        }
+        return files.toArray(new IoTDBFile[files.size()]);
+      } catch (IOException e) {
+        e.printStackTrace();
+      }
+    }
+
+    File[] rawFiles = super.listFiles();
+    IoTDBFile[] files = new IoTDBFile[rawFiles.length];
+    for (int i = 0; i < rawFiles.length; i++) {
+      files[i] = new IoTDBFile(rawFiles[i].getPath());
+    }
+    return files;
+  }
+
+  public IoTDBFile[] listFiles(FileFilter filter) {
+    if (TSFileConfig.isHdfsStorage) {
+      ArrayList<IoTDBFile> files = new ArrayList<>();
+      try {
+        PathFilter pathFilter = new GlobFilter(filter.toString()); // TODO
```

Comment:
TODO what?

Reply:
`PathFilter` / `GlobFilter` in hadoop file system needs _RegExpr_ as param to filter files, however every `FileFilter` in java file system needs to implement interface `FileFilter`. I intend to fix this method to support hadoop file in [next step](https://github.com/apache/incubator-iotdb/pull/362) since this PR just encapsulated "IoTDBFile" to ensure java file system works well...